doc/developer-guide/internal-libraries/intrusive-list.en.rst

```diff
@@ -40,16 +40,15 @@ Definition
 
    .. type:: value_type
 
-      The class for elements in the container, deduced from the return types of the link accessor methods
-      in :class:`L`.
+      The type of elements in the container, deduced from the return types of the link accessor methods
+      in :arg:`L`.
```

Comment:
* Above we have a description about how the type "`T`" of elements are deduced, but this seems incorrect even before this PR. It seems this should refer to `value_type` as well, or am I missing something?
  * If it is inconsistent, would this PR be appropriate to fix it?
* Shouldn't `L` be a type derived from an abstract class? It seems we are defining the interface for the linkage type only in English documentation and not in the code. Is that what we want?
  * (This is more a general question and not a question about this PRs change.)

Reply:
I thought I'd fixed all the 'T' to be 'value_type'. It was originally 'T' but the implementation uses 'value_type' for STL compliance. Might as well fix it here.

I'm not sure why `L` should be derived from anything. It's just a bag the static methods are put in, because that's how you pass such things to a template. Since every method depends on `value_type`, everything inherited would have to be overridden, at which point why inherit? The structure of `L` is implicitly defined by how it is used in the `IntrusiveDList` template, the text here is a human readable description of what interface is required by the code.

Reply:
It's also the case that an abstract super class would make the methods virtual, and not static, which would not work. Compiler optimizations would be disabled, whereas with the current structure the compiler can optimize away the link accessors methods entirely.

Reply:
> Might as well fix it here.

OK

> It's also the case that an abstract super class would make the methods virtual, and not static, which would not work. 

I see. I guess it is a limitation of the language then. As you mentioned in person, the current code in fact will result in compiler errors, but since they might not be straightforward errors we document what a developer needs to implement for `L` here.

Reply:
This was addressed.